core/src/test/java/org/everit/json/schema/loader/SchemaLoaderTest.java

```diff
@@ -140,6 +142,30 @@ public void integerSchema() {
         assertTrue(actual.requiresInteger());
     }
 
+    @Test
+    public void conditionalSchemaIf() {
+        ConditionalSchema actual = (ConditionalSchema) loadAsV7(get("conditionalSchemaIf"));
+        Assert.assertTrue(actual.getIfSchema().isPresent());
+        Assert.assertFalse(actual.getThenSchema().isPresent());
+        Assert.assertFalse(actual.getElseSchema().isPresent());
+    }
+
+    @Test
+    public void conditionalSchemaThen() {
+        ConditionalSchema actual = (ConditionalSchema) loadAsV7(get("conditionalSchemaThen"));
+        Assert.assertFalse(actual.getIfSchema().isPresent());
+        Assert.assertTrue(actual.getThenSchema().isPresent());
+        Assert.assertFalse(actual.getElseSchema().isPresent());
+    }
+
+    @Test
+    public void conditionalSchemaElse() {
+        ConditionalSchema actual = (ConditionalSchema) loadAsV7(get("conditionalSchemaElse"));
+        Assert.assertFalse(actual.getIfSchema().isPresent());
+        Assert.assertFalse(actual.getThenSchema().isPresent());
+        Assert.assertTrue(actual.getElseSchema().isPresent());
+    }
```

Comment:
Also it would be good to have tests for
 - checking that the if-then-else schema is loaded only in DRAFT_7 mode
 - the case where a subschema is a boolean (eg, `{"if" : true }` )
 - a case where all 3 keywords are present